test/unit/route53-client.server.test.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  createHostedZone,
+  executeChangeSet,
+  getDnsRecordSetPage,
+} from 'services/reconciler/route53-client.server';
+
+describe('Route53 Client functions test', () => {
+  let hostedZoneId: string;
+  const rootDomain = 'starchart.com';
+  const username = 'jdo12';
+  const fqdn = (subdomain: string = String(Date.now())) => `${subdomain}.${username}.${rootDomain}`;
+
+  const resetHostedZone = () => {
+    try {
+      fetch('http://localhost:5053/moto-api/reset', {
+        method: 'POST',
+      });
+    } catch (error) {
+      throw new Error(`Error occurred while deleting records in hosted zone: ${error}`);
+    }
+  };
+
+  beforeAll(async () => {
+    hostedZoneId = await createHostedZone(rootDomain);
+    process.env.AWS_ROUTE53_HOSTED_ZONE_ID = hostedZoneId;
+    process.env.ROOT_DOMAIN = rootDomain;
+  });
+
+  afterAll(() => resetHostedZone());
```

Comment:
@humphd 
I usually try and do the cleanup in the beforeAll section (before initialization):
If a previous test has failed, was interrupted (ctrl+c), or by any other source, there is some garbage in the system, that the test may or may not be able to tolerate, I want to clean that out before I run the test, so the test can truley be atomic (and not depend on the previous test cleaning up after itself)

Just a suggestion

Reply:
I tried that but trashing the hosted zone while while the modules are loaded seems to cause tests to fail.